src/components/common/ProtocolQuery/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { MouseEvent, useEffect, useState } from 'react';
```

Comment:
if this represents a GroupQuery e,g byName, but later we may add criteria like an array of addresses and exclusion, shouldnt it be named GroupQuery?

You can have a query with 3 protocols not just one